File: TreeView/TreeViewReadOnlyCheckboxes/RadTreeView_ReadOnly_CS/RadReadOnlyTreeView.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Telerik.WinControls.UI;
using System.Collections.ObjectModel;

namespace RadTreeView_ReadOnly_CS
{
    class RadReadOnlyTreeView : RadTreeView
    {

        private bool m_ReadOnly = false;

        public RadReadOnlyTreeView()
        {
            base.NodeFormatting +=new TreeNodeFormattingEventHandler(RadReadOnlyTreeView_NodeFormatting);
        }

        public override string ThemeClassName  
        { 
            get 
            {  
                return typeof(RadTreeView).FullName;  
            }  
        }  

        void RadReadOnlyTreeView_NodeFormatting(object sender, TreeNodeFormattingEventArgs e)
        {
            if (base.CheckBoxes)
            {
                if (m_ReadOnly)
                {
                    ((RadCheckBoxElement)e.NodeElement.Children[2]).Enabled = false;
                }
                else
                {
                    ((RadCheckBoxElement)e.NodeElement.Children[2]).Enabled = true;
                }
            }
        }

        [System.ComponentModel.Browsable(true)] 
        [System.ComponentModel.DefaultValue(false)] 
        [System.ComponentModel.Category("Behavior")]
        [System.ComponentModel.Description("If the tree view has checkboxes, then allows the treeview checkboxes to be read only")]
        public bool CheckBoxReadOnly
        {
            get
            {
                return m_ReadOnly;
            }
            set
            {
                m_ReadOnly = value;
            }
        }

    }
}

File: TreeView/TreeViewReadOnlyCheckboxes/RadTreeView_ReadOnly_VB/RadReadOnlyTreeView.vb
Imports Telerik.WinControls.UI
Imports System.Collections.ObjectModel


Public Class RadReadOnlyTreeView
    Inherits RadTreeView

    Private m_ReadOnly As Boolean = False

    Public Overrides Property ThemeClassName() As String
        Get
            Return GetType(RadTreeView).FullName
        End Get
        Set(ByVal value As String)
        End Set
    End Property

    Private Shadows Sub NodeFormatting(sender As System.Object, e As Telerik.WinControls.UI.TreeNodeFormattingEventArgs) Handles MyBase.NodeFormatting
        If MyBase.CheckBoxes Then
            If m_ReadOnly Then
                CType(e.NodeElement.Children(2), RadCheckBoxElement).Enabled = False
            Else
                CType(e.NodeElement.Children(2), RadCheckBoxElement).Enabled = True
            End If
        End If
    End Sub

    <System.ComponentModel.Browsable(True)> _
    <System.ComponentModel.DefaultValue(False)> _
    <System.ComponentModel.Category("Behavior")> _
    <System.ComponentModel.Description("If the tree view has checkboxes, then allows the treeview checkboxes to be read only")> _
    Public Property CheckBoxReadOnly() As Boolean
        Get
            Return m_ReadOnly
        End Get
        Set(value As Boolean)
            m_ReadOnly = value
        End Set
    End Property

End Class
